include/unifex/connect_awaitable.hpp

```diff
@@ -182,31 +176,17 @@ namespace _await_cpo {
         // for the receiver to destroy the coroutine.
         co_yield [&](result_type&& result) {
               return [&] {
-                constexpr size_t valueOverloadCount =
-                    sender_value_types_t<Awaitable, count_types, single_value_type>::value;
-                static_assert(valueOverloadCount <= 1);
-
-                if constexpr (valueOverloadCount == 1) {
-                  constexpr size_t valueCount =
-                      sender_value_types_t<Awaitable, type_identity_t, count_types>::value;
-                  if constexpr (valueCount == 0) {
-                    unifex::set_value(std::move(receiver));
-                  } else if constexpr (valueCount == 1) {
-                    unifex::set_value(std::move(receiver), static_cast<result_type&&>(result));
-                  } else {
-                    std::apply(set_value_applicator<Receiver>{receiver}, (result_type&&)result);
-                  }
+                if constexpr (std::is_void_v<await_result_t<awaitable_type>>) {
```

Comment:
What do you think of checking for `unifex::same_as<result_type, unit>` instead?

I think the difference would be that an `operator co_await` that directly returns `unifex::unit` would be collapsed to no value instead of passed along.  Dunno if that's a win or if it would increase confusion.

Reply:
I dunno either. I'd rather not make an unrelated change like that in this PR.